### DISCUSSION TITLE
Grouping multiple rows and showing joins as inner objects

### DISCUSSION THREAD

**Discussion by lazaroofarrill** [supabase/supabase#12000] [Questions] [answered]
I have the following view

```sql
create view business_view as
select b.id,
       b.name,
       b.info,
       ROUND(AVG(br.rating), 1) as ratings,
       bc.id                    as category_id,
       bc.name                  as category_name
from businesses as b
         left join business_ratings as br on br.business_id = b.id
         left join business_business_category bbc on bbc.business_id = b.id
         left join business_categories bc on bc.id = bbc.category_id
group by b.id, bc.id
order by b.id;
```

I'm receiving the following response
```json
[
{
    "id": "1"
    "category_id": "cat1",
    "category_name": "name"
},
{
    "id": "1"
    "category_id": "cat2",
    "category_name": "cat2"
}
]
```

However I wont to group this to rows like this:

```json
[
{
    "id": "1",
    "category":[
      {
        "id": "cat1",
       "name": "cat1"
    }, {
        "id": "cat2",
       "name": "cat2"
    }
    ]
}
]
```

Is this currently achievable in supabase?

**Comment by GaryAustin1** [comment, marked as answer]
This is not really a Supabase question but a Postgres question.   Someone here might have an answer for you but you should be researching Postgres solutions on the web and not just Supabase.

**Reply by lazaroofarrill** [reply]
> This is not really a Supabase question but a Postgres question.   Someone here might have an answer for you but you should be researching Postgres solutions on the web and not just Supabase.

I don't see how this is a Postgres question unflattenning data usually occurs at the orm level since sql only returns rows.

I got to achieve this using the select query param as per PostgRest docs but I was hoping to   not have to put logic into my urls.
